### PR TITLE
use HtPP, HtUP, HtUU

### PR DESCRIPTION
**Pull request by ret2libc**

### CODE REVIEW

File: libr/anal/state.c [hunk 1]
```diff
@@ -51,25 +48,19 @@ R_API void r_anal_state_insert_bb(RAnalState* state, RAnalBlock *bb) {
 	if (!r_anal_state_search_bb (state, bb->addr) && state->current_fcn) {
 		r_list_append (state->current_fcn->bbs, bb);
 		state->bytes_consumed += state->current_bb->op_sz;
-		const char *key = sdb_fmt ("0x%08"PFMT64x, bb->addr);
-		if (!ht_insert (state->ht, key, bb)) {
+		if (!ht_up_insert (state->ht, bb->addr, bb)) {
```
Comment (radare) [inline libr/anal/state.c]: as long as now we have multiple type hashtables maybe good to add this `up` in the variable name holding the reference of the ht instance. So its easier for the user to know which kv types are expected to be used with each ht.

Comment (ret2libc) [reply libr/anal/state.c]: we could, but do we want to? in general we don't specify the type in the name (e.g. we don't do "name_s", "bytes_ut64", etc.) and the fact that we have strongly typed hashtables (HtUP, HtPP and HtUU are very different types now) makes it impossible to mix them up.

**Comment by radare** [comment]
We can ignore this change i guess because it will spit compile time warnings if we use the wrong apis right?

**Comment by ret2libc** [comment]
> We can ignore this change i guess because it will spit compile time warnings if we use the wrong apis right?

Yes, it's right, it will gives warning likes:
```
../libr/anal/state.c: In function ‘r_anal_state_insert_bb’:
../libr/anal/state.c:51:27: warning: passing argument 1 of ‘ht_pp_insert’ from incompatible pointer type [-Wincompatible-pointer-types]                                                      
   if (!ht_pp_insert (state->ht, bb->addr, bb)) {
                      ~~~~~^~~~
In file included from ../libr/include/sdb/ht_pp.h:9,                                                                                                                                         
                 from ../libr/include/sdb/sdbht.h:4,
                 from ../libr/include/sdb/sdb.h:14,
                 from ../libr/include/sdb.h:1,
                 from ../libr/include/r_list.h:6,
                 from ../libr/include/r_io.h:6,
                 from ../libr/include/r_anal.h:14,
                 from ../libr/anal/state.c:3:
../libr/include/sdb/ht_inc.h:92:39: note: expected ‘HtPP *’ {aka ‘struct ht_pp_t *’} but argument is of type ‘HtUP *’ {aka ‘struct ht_up_t *’}                                               
 SDB_API bool Ht_(insert)(HtName_(Ht)* ht, const KEY_TYPE key, VALUE_TYPE value);
                                       ^                                                                                                                                                     
../libr/anal/state.c:51:35: warning: passing argument 2 of ‘ht_pp_insert’ makes pointer from integer without a cast [-Wint-conversion]                                                       
   if (!ht_pp_insert (state->ht, bb->addr, bb)) {
                                 ~~^~~~~~
In file included from ../libr/include/sdb/ht_pp.h:9,
                 from ../libr/include/sdb/sdbht.h:4,
                 from ../libr/include/sdb/sdb.h:14,
                 from ../libr/include/sdb.h:1,
                 from ../libr/include/r_list.h:6,
                 from ../libr/include/r_io.h:6,
                 from ../libr/include/r_anal.h:14,
                 from ../libr/anal/state.c:3:
../libr/include/sdb/ht_inc.h:92:58: note: expected ‘const void *’ but argument is of type ‘long long unsigned int’
 SDB_API bool Ht_(insert)(HtName_(Ht)* ht, const KEY_TYPE key, VALUE_TYPE value);
```

**Comment by radare** [comment]
then i'm all for merging this (with a capitalized commit message)

**Pull request merged**